### PR TITLE
Blog App: validations, model specs, and n+1 problems

### PR DESCRIPTION
**Pull request by toliboff**
### In this PR were implemented following:
- Add the following validations:
  - For the User model:
    - Name must not be blank.
    - PostsCounter must be an integer greater than or equal to zero.
  - For the Post model:
    - Title must not be blank.
    - Title must not exceed 250 characters.
    - CommentsCounter must be an integer greater than or equal to zero.
    - LikesCounter must be an integer greater than or equal to zero.
- Add unit specs for all of your models' methods and validations.
- Add flash messages in the create actions in all your controllers.
- Make sure that the N+1 problem is solved when fetching all posts and their comments for a user.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -3,6 +3,8 @@ class User < ApplicationRecord
   has_many :likes, dependent: :destroy, foreign_key: 'author_id'
   has_many :comments, dependent: :destroy, foreign_key: 'author_id'
 
+  validates :name, presence: true
+  validates :posts_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
   def recent_posts
     posts.last(3)
   end
```
Comment (AshakaE) [inline app/models/user.rb:10]: - [x] Please write a unit test for this method

File: app/models/post.rb [hunk 2]
```diff
@@ -3,6 +3,10 @@ class Post < ApplicationRecord
   has_many :likes
   has_many :comments
 
+  validates :title, presence: true, length: { maximum: 250 }
+  validates :comments_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
+  validates :likes_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
+
   def last_comments
     comments.last(5)
   end
```
Comment (AshakaE) [inline app/models/post.rb:12]: - [x] Please write a unit test for this method

**Review by AshakaE — CHANGES_REQUESTED** [review]
### Hi @toliboff  🙋‍♀️, 

Good job so far! 👌
Pull request ✔️ 
README ✔️ 
Git Flow ✔️ 

There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


#  Required Changes ♻️ 🚧
_Check the comments under the review._

###  Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Lameck1 — APPROVED** [review]
Hi, @toliboff,

You've done well working on the previous review concerns 👏🏼.
Your project is now complete!  consider merging this branch :shipit:

Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by toliboff** [comment]
Thank you

**Pull request merged**